docs/pages/reference/architecture/relay.mdx

```diff
@@ -0,0 +1,67 @@
+
+This alternate connectivity is beneficial in situations where the client and the resource are known to be close together in a physical or logical sense (the same data center, office, campus, geographical region) and regular connections through the control plane would incur higher latency, lower throughput, or higher costs.
+
+Unlike the Proxy service, the Relay does not host a web UI, intercept connections, impersonate users or provide access to the Teleport control plane API. As such, it's simpler to deploy and keep secure, and it's thus suitable to be deployed even in environments where it wouldn't be possible to securely deploy a Proxy instance.
+
+<Admonition type="warning">
+
+The Teleport Relay service is intended for specific scenarios where clients and agents are in the same network segment and there is a need for connectivity that does not go through the Teleport control plane. It is not a required or recommended cluster component in most Teleport deployments.
+
+</Admonition>
+
+## How it works
+
+A Relay service deployment consists of one or more Relay instances configured with the same Relay group name, reachable by Teleport agents and clients through a L4 network load balancer. Relay instances of the same group receive connections from agents and clients through the load balancer, and can connect to each other directly. Relay instances listen on two separate ports to serve connections from agents and clients, and it's possible to use different load balancers and hostnames if necessary. The typical setup uses a single load balancer serving the two ports, reachable at a single hostname, with port 3042 used by agents and port 443 used by clients.
+
+An agent can be configured to use a given Relay by adding the address of the load balancer to its `teleport.yaml` configuration file:
+
+```yaml
+teleport:
+  proxy_server: proxy.example.com:443
+  relay_server: relay.example.com:3042
+```
+
+Agents running the SSH service that are also connected to a Proxy (i.e. they're running in "tunnel mode") will also open tunnels to the Relay service. The address specified in the configuration should resolve to the load balancer for the port used by agents.
+
+When an agent is configured to open tunnels to a given Relay group, it will periodically check the Relay configuration advertised by the Relay instances, then open enough tunnel connections to distinct Relay instances as determined by the Relay group configuration. The target connection count in the configuration should be not bigger than the amount of active and reachable Relay instances in the group at any given time, or all agents configured to use the group will keep connecting to the load balancer to reach for distinct instances that are not actually available.
```

Comment:
This section references a number of Relay configuration options that have not yet been introduced. Do you think it would be worthwhile to include an example Relay configuration with high level explanation of the fields before diving into this section? Or linking to the reference section for the Relay service?